crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -28,27 +33,35 @@ impl Dex {
         Self {
             dex,
             slippage: config.slippage,
+            concurrent_requests: config.concurrent_requests,
             fills: Fills::new(config.smallest_partial_fill),
         }
     }
 
     pub async fn solve(&self, auction: auction::Auction) -> Vec<solution::Solution> {
-        // TODO:
-        // * concurrency
-
         let mut solutions = Vec::new();
         let solve_orders = async {
-            for order in auction.orders.iter().filter_map(order::UserOrder::new) {
-                let span = tracing::info_span!("solve", order = %order.get().uid);
-                if let Some(solution) = self
-                    .solve_order(order, &auction.tokens, auction.gas_price)
-                    .instrument(span)
-                    .await
-                {
-                    solutions.push(solution);
-                }
+            // Collect user orders into a vector for chunking. Note that we
```

Comment:
I think you are reinventing [`futures::stream::buffered()`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffered) here.
`buffered()` would be a bit nicer because we are always saturating the max number of active parallel requests instead of having the whole next batch being blocked by a single slow request.

Reply:
I was, and this was a great suggestion.

I wrestled with the Rust compiler for a long time, but got it to work.